freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -70,15 +68,24 @@ def handle(self, event):
             self.log_info(msg)
             return []
 
-        # Get original nvrs of all builds in the advisory
-        original_nvrs = [build.original_nvr for build in db_event.builds]
+        # Get builds NVRs from the advisory attached to the message/event and
+        # then get original NVR for every build
+        original_nvrs = set()
+        for product_info in event.advisory.builds.values():
+            for build in product_info['builds']:
+                # Search for the first build that triggered the chain of rebuilds
+                # for every shipped NVR to get original NVR from it
+                original_nvr = ArtifactBuild.get_original_nvr_by_rebuilt_nvr(
```

Comment:
I don't think this class method is necessary, we can just filter `ArtifactBuild` with the `rebuild_nvr` directly

Reply:
I can move it from the class but I would at least make a separate method for it, just to not overextend the `handle()` method. And why it's a class method, I thought that all logic including querying particular tables should be in `models.` class of that table, as in `models.Event` for example

Reply:
@qixiang so you think it's better to move it outside of `ArtifactBuild` class?

Reply:
If we can just get the original nvr with one line query filter, it doesn't worth to have separate class method, now I'm thinking about the case of looking for the orignal nvr recursively, I'll add comment in the other thread.